realm/realm-library/src/androidTest/java/io/realm/BulkInsertTests.java

```diff
@@ -317,7 +319,33 @@ public void insertToRealm_NullPrimaryKey() {
     }
 
     @Test
-    public void insertOrUpdateToRealm() {
+    public void insert_duplicatedPrimaryKeyFails() {
+
+        // Single object with 2 references to two objects with the same ID
+        AllJavaTypes obj = new AllJavaTypes(2);
+        obj.setFieldList(new RealmList<AllJavaTypes>(new AllJavaTypes(1), new AllJavaTypes(1)));
+        realm.beginTransaction();
+        try {
+            realm.insert(obj);
+            fail();
+        } catch (RealmPrimaryKeyConstraintException ignored) {
```

Comment:
Why would this throw? It is clearly valid that `RealmList` can have duplicated elements. I would say if use have two objects with same pk but different other fields, then it is an undefined behaviour and user should not do that.